### PR TITLE
Adds a Himeo voidsuit.

### PR DESCRIPTION
**Pull request by TheStryker**
Adds a Himeo voidsuit kit to the loadout that can be used to reskin a mining voidsuit into a Himeo voidsuit and back. Mechanically, it is identical to a mining suit.
HIMEO

### CODE REVIEW

**Comment by TheStryker** [comment]
!review

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -1353,3 +1353,49 @@ var/list/total_extraction_beacons = list()
...
+	desc_fluff = "As part of a cost-cutting and productivity-enhancing initiative, NanoTrasen has authorized a number of Himean Type-76 'Fish Fur'\
+	for use by miners originating from the planet. Most of these suits are assembled in Cannington and painstakingly optimized on-site by their\
+	individual operator leading to a large trail of red tape as NanoTrasen is forced to inspect these suits to ensure their safety."
+	desc_info = "In order to convert a mining voidsuit into a Himean voidsuit, simply click on this box with a voidsuit or helmet in hand.\
+	The same process can be used to convert a Himean voidsuit back into a regular voidsuit. Make sure not to have a helmet or tank in the suit\
+	or else it will be deleted."
+	w_class = ITEMSIZE_SMALL
+
+/obj/item/himeo_kit/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/clothing/suit/space/void/mining))
```
Comment (Geevies) [inline code/modules/mining/mine_items.dm:1374]: these should eject any connect tanks, helmets, and magboots before deleting

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -1353,3 +1353,49 @@ var/list/total_extraction_beacons = list()
...
+		var/obj/item/clothing/suit/space/void/mining/himeo/P = new /obj/item/clothing/suit/space/void/mining/himeo(user.loc)
+		user.put_in_hands(P)
+	if(istype(W, /obj/item/clothing/head/helmet/space/void/mining))
+		user.drop_item(W)
+		qdel(W)
+		to_chat(user, SPAN_NOTICE("Your permit for a Himean voidsuit helmet has been processed. Enjoy!"))
+		playsound(src.loc, 'sound/weapons/blade_open.ogg', 50, 1)
+		var/obj/item/clothing/head/helmet/space/void/mining/himeo/P = new /obj/item/clothing/head/helmet/space/void/mining/himeo(user.loc)
+		user.put_in_hands(P)
+	if(istype(W, /obj/item/clothing/suit/space/void/mining/himeo))
```
Comment (Geevies) [inline code/modules/mining/mine_items.dm:1388]: ditto

**Review by Geevies — CHANGES_REQUESTED** [review]

File: code/modules/mining/mine_items.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/mining/mine_items.dm:1374]: and also any attached accesories

**Review by Geevies — APPROVED** [review]

File: code/modules/mining/mine_items.dm [hunk 3]
```diff
@@ -1353,3 +1353,88 @@ var/list/total_extraction_beacons = list()
...
+			H.canremove = TRUE
+			H.forceMove(get_turf(src))
+			M.helmet = null
+		var/obj/item/tank/T = M.tank
+		if(T)
+			T.canremove = TRUE
+			T.forceMove(get_turf(src))
+			M.tank = null
+		for(var/obj/accessory in M.accessories)
+			M.remove_accessory(user, accessory)
```
Comment (skull132) [inline code/modules/mining/mine_items.dm:1392]: This touching `canremove` smells dubious. _Very_ dubious. This paint kit likely should not work on voidsuits which cannot be disassembled (for whatever reason). And as the code is constructed at the moment, that is possible. So basically, leave the deconstruction of the voidsuit up to the person. And just error out when there are pieces still attached to it.

File: code/modules/mining/mine_items.dm [hunk 4]
```diff
@@ -1353,3 +1353,88 @@ var/list/total_extraction_beacons = list()
...
+		user.drop_item(W)
+		qdel(W)
+		to_chat(user, SPAN_NOTICE("Your permit for a Himean voidsuit has been processed. Enjoy!"))
+		playsound(src.loc, 'sound/weapons/blade_open.ogg', 50, 1)
+		var/obj/item/clothing/suit/space/void/mining/himeo/P = new /obj/item/clothing/suit/space/void/mining/himeo(user.loc)
+		user.put_in_hands(P)
+
+	if(istype(W, /obj/item/clothing/head/helmet/space/void/mining))
```
Comment (skull132) [inline code/modules/mining/mine_items.dm:1400]:
```suggestion
	else if(istype(W, /obj/item/clothing/head/helmet/space/void/mining))
```

File: code/modules/mining/mine_items.dm (hunk 1 above)
Comment (skull132) [reply code/modules/mining/mine_items.dm:1374]: Heavy disagree. It should error if it attempts to repaint such a voidsuit. Allowing it to remove attachments and the like will result in it dismantling things which are not meant to be dismantled.

**Review by skull132 — CHANGES_REQUESTED** [review]

**Review by Alberyk — APPROVED** [review]

**Review by fernerr — APPROVED** [review]

**Review by skull132 — APPROVED** [review]

**Pull request merged**